projects/plugins/boost/app/assets/src/js/pages/settings/elements/CriticalCssErrorDescription.svelte

```diff
@@ -28,27 +30,44 @@
 	export let showClosingParagraph = true;
 
 	/**
-	 * @member {ErrorSet} errorSet Error Set to display a description of, from a Recommendation or CriticalCssStatus.
+	 * A set of errors to display recommendations from, from a Recommendation or CriticalCssStatus.
 	 */
-	export let errorSet;
+	export let errorSet: ErrorSet;
 
+	type FormattedURL = {
+		/**
+		 * The URL to display in the list.
+		 */
+		href: string;
+		/**
+		 * The URL to link to.
+		 */
+		label: string;
+	};
 	// Keep a set of URLs in an easy-to-render {href:, label:} format.
 	// Each should show the URL in its label, but actually link to error.meta.url if available.
-	let displayUrls = [];
-	$: displayUrls = Object.entries( errorSet.byUrl ).map( ( [ url, error ] ) => ( {
-		href: error.meta.url ? error.meta.url : url,
-		label: url,
-	} ) );
+	let displayUrls: FormattedURL[] = [];
+
+	$: displayUrls = Object.entries( errorSet.byUrl ).map( ( [ url, error ] ) => {
+		let href = url;
+		if ( error.meta.url && typeof error.meta.url === 'string' ) {
+			href = error.meta.url;
+		}
+		return {
+			href,
+			label: url,
+		};
+	} );
 
-	const templateVars = {
+	const templateVars: TemplateVars = {
 		...actionLinkTemplateVar( () => dispatch( 'retry' ), 'retry' ),
 		...supportLinkTemplateVar(),
 	};
 </script>
 
 <div class="jb-critical-css__error-description">
 	<span class="error-description">
-		<TemplatedString template={describeErrorSet( errorSet )} vars={{ templateVars }} />
+		<TemplatedString template={describeErrorSet( errorSet )} vars={templateVars} />
```

Comment:
Was it broken previously?

Reply:
I don't know 🤷 

The component exports were defined like this:


```svelte
	/**
	 * Template vars to replace in the template. Each key corresponds to a
	 * <pseudo-html /> tag name to replace, each value should be an array with
	 * two entries; the HTML tag type, and an object full of attributes.
	 *
	 * @member {TemplateVars} vars
	 */
	export let vars;

	```
When I added `TemplateVars` type, TypeScript complained about an unexpected object because `{{ }}` creates an object.
<img width="1205" alt="CleanShot 2022-10-26 at 16 03 02@2x" src="https://user-images.githubusercontent.com/988095/198033451-52839755-593c-4794-97c0-546b4b9ee747.png">